pkg/schemapatcher/gen.go

```diff
@@ -73,6 +73,16 @@ type Generator struct {
 	// closest sentence boundary if it exceeds n characters.
 	MaxDescLen *int `marker:",optional"`
 
+	// AllowDangerousTypes allows types which are usually omitted from CRD generation
+	// because they are not recommended.
+	//
+	// Currently the following additional types are allowed when this is true:
+	// float32
+	// float64
+	//
+	// Left unspecified, the default is false
+	AllowDangerousTypes *bool `marker:",optional"`
```

Comment:
Is there a use case on why we'd allow floats to be used in CRDs? I understand there is a flag you can set, although these have so many drawbacks that I'm a bit hesitant to allow

Reply:
@vincepri 

If floats are not allowed to be used in CRDs, it is better to remove the `allowDangerousTypes` option from `controller-gen crd`. 

Since users are allowed to use floats in CRDs by running `controller-gen crd:allowDangerousTypes=true`, we should also allow `controller-gen schemapatch:allowDangerousTypes=true` for consistency.

If there are projects that already have floats in CRDs, users may need to:
1. Switch from manually writing CRDs to using `controller-gen schemapatch` for automated  OpenAPI Schema generation.
2. Switch from using `controller-gen crd` to using `controller-gen schemapatch` for updating only the OpenAPI Schema.

Reply:
I would move for discouraging the dangerous types. Kube API conventions mention not to use floats because they don't round trip well and different languages and libraries for Kube don't handle floats in the same way. If we don't already support dangerous types as part of the controller tools project, do we really want to enable it? That would be encouraging it IMO

Reply:
@JoelSpeed So It would be better to open another PR for removing the `allowDangerousTypes` option from `controller-gen crd`?

Reply:
You can't do that, it would be a breaking change to `controller-gen`, but I think we have the choice not to extend the support and include this new addition, since that would make it easier to use the dangerous types, and therefore encourage their use.

If I were starting this from scratch then sure, I wouldn't have added it, but now that it's there, we must support it.